### PR TITLE
Set the admin setting when options are available.

### PR DESCRIPTION
**Pull request by SergioEstevao**
How to test:

 * Add a self-hosted site where you have admin rights
 * Check the Blog->Blog Details->Settings is available

 * Add a self-hosted site where you don't admin rights, just editor rights for example
 * Check if the Blog->Blog Details->Settings is not available

Needs Review: @jleandroperez

### CODE REVIEW

**Comment by jleandroperez** [comment]
Hello there @SergioEstevao!

I've found a (minor) glitch:

-- 

1. Fresh install WPiOS
2. Log into a self hosted blog (admin)
3. Open `My Sites` and tap over the site

The `Settings` row won't be initially visible (because the options are being sync'ed). Hitting Back + tapping again over the site shows the extra row.

--

Second glitch is: the `Settings` row is showing up for a user that's a `Subscriber`. I'm assuming that's not expected.

File: WordPress/Classes/Services/BlogService.m [hunk 1]
```diff
@@ -640,6 +640,7 @@ - (OptionsHandler)optionsHandlerWithBlogObjectID:(NSManagedObjectID *)blogObject
                                                                            error:nil];
             if (blog) {
                 blog.options = [NSDictionary dictionaryWithDictionary:options];
+                blog.isAdmin = YES;
```
Comment (jleandroperez) [inline WordPress/Classes/Services/BlogService.m]: Shouldn't we be parsing this from somewhere? (Otherwise everyone will be an admin, unless i'm missing a huge something!)

**Comment by SergioEstevao** [comment]
@jleandroperez ready for another round of review.

File: WordPress/Classes/Services/BlogService.m [hunk 2]
```diff
@@ -638,22 +638,33 @@ - (OptionsHandler)optionsHandlerWithBlogObjectID:(NSManagedObjectID *)blogObject
...
                 }
-
-                [[ContextManager sharedInstance] saveContext:self.managedObjectContext];
+                return;
+            }
+            blog.options = [NSDictionary dictionaryWithDictionary:options];
+            //HACK:Sergio Estevao (2015-08-31): Because there is no direct way to
+            // know if a user has permissions to change the options we check if the blog title property is read only or not.
+            if (blog.options[@"blog_title"] &&
+                blog.options[@"blog_title"][@"readonly"]) {
```
Comment (jleandroperez) [inline WordPress/Classes/Services/BlogService.m]: (Just in case) Maybe switch over to Safe Expectations, for peace of mind?

```
[[blog.options dictionaryForKey:@"blog_title"] numberForKey:@"readonly"]
```

File: WordPress/Classes/Services/BlogService.m [hunk 3]
```diff
@@ -638,22 +638,33 @@ - (OptionsHandler)optionsHandlerWithBlogObjectID:(NSManagedObjectID *)blogObject
...
+                blog.isAdmin = ![blog.options[@"blog_title"][@"readonly"] boolValue];
             }
+            float version = [[blog version] floatValue];
+            if (version < [MinimumVersion floatValue]) {
+                if (blog.lastUpdateWarning == nil
+                    || [blog.lastUpdateWarning floatValue] < [MinimumVersion floatValue])
+                {
+                    // TODO :: Remove UI call from service layer
```
Comment (jleandroperez) [inline WordPress/Classes/Services/BlogService.m]: Maybe move this snippet over to another PR, and place it in the right spot?

Comment (jleandroperez) [inline WordPress/Classes/Services/BlogService.m]: Ahhhh sorry, my bad, thought this was new code! just realized this is actually old code moved over.

File: WordPress/Classes/Services/BlogSyncFacade.m [hunk 4]
```diff
@@ -49,6 +49,12 @@ - (void)syncBlogWithUsername:(NSString *)username
      */
     blog.password = password;
     blog.options = options;
+    //HACK:Sergio Estevao (2015-08-31): Because there is no direct way to
+    // know if a user has permissions to change the options we check if the blog title property is read only or not.
+    if (blog.options[@"blog_title"] &&
+        blog.options[@"blog_title"][@"readonly"]) {
```
Comment (jleandroperez) [inline WordPress/Classes/Services/BlogSyncFacade.m]: Same as above, maybe Safe Expectations, for peace of mind?

File: WordPress/Classes/ViewRelated/Blog/SiteSettingsViewController.m [hunk 5]
```diff
@@ -111,13 +111,24 @@ - (void)viewDidLoad
...
-    if ([self.blog supports:BlogFeatureRemovable]) {
-        self.tableSections = [self.tableSections arrayByAddingObject:@(SiteSettingsSectionRemoveSite)];
+        if (self.blog.account) {
+            self.tableSections = @[@(SiteSettingsSectionGeneral)];
+        } else {
+            self.tableSections = @[@(SiteSettingsSectionGeneral), @(SiteSettingsSectionAccount)];
+        }
+        if ([self.blog supports:BlogFeatureRemovable]) {
+            self.tableSections = [self.tableSections arrayByAddingObject:@(SiteSettingsSectionRemoveSite)];
+        }
```
Comment (jleandroperez) [inline WordPress/Classes/ViewRelated/Blog/SiteSettingsViewController.m]: Since both branches of this `if` look roughly the same, and we're adding sections (as / if required), maybe something like this?


```
NSMutableArray *tableSections = [NSMutableArray arrayWithObject:@(SiteSettingsSectionGeneral)];

if (!self.blog.account) {
   [tableSections addObject:@(SiteSettingsSectionAccount)];
}

if (self.blog.isAdmin) {
   [tableSections addObject:@(SiteSettingsSectionWriting)];
}

if ([self.blog supports:BlogFeatureRemovable]) {
    [tableSections addObject:@(SiteSettingsSectionRemoveSite)];
}
```

**Comment by jleandroperez** [comment]
Just tested the following scenarios:

- Self Hosted: 'Mortal' User
- Self Hosted: Admin User
- WordPress.com

Looking great! :shipit:

**Comment by koke** [comment]
I believe this still has a problem: since the option to remove a site now lives inside site settings, there's no way to remove a self hosted site from the app if you are not an admin on the site. I think the 'Settings' item should always be available and we should decide what's on the Settings screen based on capabilities.

**Comment by jleandroperez** [comment]
@koke Sergio updated the code, so that the appropiate `Settings` are not displayed (or are not editable) when applicable. Sorry about the fuzz, i think the PR description is just outdated, but it works exactly as you've outlined.
